src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/PasswordResetTokenService.java

```diff
@@ -0,0 +1,370 @@
+package de.terrestris.shogun2.service;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.UUID;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.log4j.Logger;
+import org.hibernate.criterion.Criterion;
+import org.hibernate.criterion.Restrictions;
+import org.hibernate.criterion.SimpleExpression;
+import org.joda.time.DateTime;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.web.util.UriUtils;
+
+import de.terrestris.shogun2.model.User;
+import de.terrestris.shogun2.model.token.PasswordResetToken;
+import de.terrestris.shogun2.util.application.Shogun2ServletContext;
+import de.terrestris.shogun2.util.mail.MailPublisher;
+
+/**
+ *
+ * @author Daniel Koch
+ *
+ */
+@Service("passwordResetTokenService")
+public class PasswordResetTokenService extends AbstractCrudService<PasswordResetToken> {
+
+	/**
+	 * The Logger
+	 */
+	private static final Logger LOG =
+			Logger.getLogger(PasswordResetTokenService.class);
+
+	/**
+	 *
+	 */
+	@Autowired
+	private Shogun2ServletContext contextUtil;
+
+	/**
+	 *
+	 */
+	@Autowired
+	private UserService userService;
+
+	/**
+	 *
+	 */
+	@Autowired
+	private MailPublisher mailPublisher;
+
+	/**
+	 * The autowired PasswordEncoder
+	 */
+	@Autowired
+	private PasswordEncoder passwordEncoder;
+
+	/**
+	 *
+	 */
+	@Autowired
+	private SimpleMailMessage resetPasswordMailMessageTemplate;
+
+	/**
+	 *
+	 */
+	@Autowired
+	private String changePasswordPath;
+
+	/**
+	 *
+	 * @param user
+	 * @return
+	 */
+	public PasswordResetToken findByUser(User user) {
+
+		SimpleExpression eqEmail = Restrictions.eq("user", user);
+		PasswordResetToken passwordResetToken =
+				dao.findByUniqueCriteria(eqEmail);
+
+		return passwordResetToken;
+	}
+
+	/**
+	 *
+	 * @return
+	 */
+	public PasswordResetToken findByIdAndToken(int id, String token) {
+
+		Criterion criteria = Restrictions.and(
+				Restrictions.eq("id", id),
+				Restrictions.eq("token", token)
+		);
+
+		PasswordResetToken passwordResetToken =
+				dao.findByUniqueCriteria(criteria);
+
+		return passwordResetToken;
+	}
+
+	/**
+	 *
+	 * @param request
+	 * @param email
+	 * @return
+	 * @throws UsernameNotFoundException
+	 * @throws Exception
+	 * @throws URISyntaxException
+	 */
+	public Boolean sendResetPasswordMail(HttpServletRequest request, String email) throws
+			UsernameNotFoundException, Exception, URISyntaxException {
```

Comment:
If possible remobve the `Exception`. I don't like the generic type… others may disagree